crates/extensions/c8y_mapper_ext/src/error.rs

```diff
@@ -149,6 +152,9 @@ pub enum CumulocityMapperError {
     #[error(transparent)]
     FromEntityStore(#[from] tedge_api::entity_store::Error),
 
+    #[error(transparent)]
+    FromEntityCacheError(#[from] crate::entity_cache::Error),
+
```

Comment:
Why do we have to maintain two error types with the same error cases?

Reply:
The errors emitted by the entity cache would be similar, but a much smaller set compared to the entity store, as it maintains far less metadata. That's why I didn't want them to share the same error type. Eventually I see the entity store structure moving from `tedge_api` to `tedge_agent` and the entity cache being mainained by the `tedge_mapper`. So, avoiding any direct dependency between them would be better for that move.